### ISSUE TITLE
Domains: registered domain displays as available for registration

### ISSUE THREAD

**Issue by meremagee** [Automattic/wp-calypso#7881]
Ran into the following while trying to map a domain I have registered on GoDaddy to a new WordPress.com site:

```
[06-Sep-2016 17:03:07 UTC] [36b1-26132] array (
  'is_registrable' => true,
  'is_available' => false,
  'is_mappable' => true,
)
```

Domain details:
- publicly registered on 9/1/16
- nameservers mapped to wp.com nameservers
- not connected to any sites previously

#### Steps that triggered, though I have not been able to repro with another domain
1. Starting at URL: wordpress.com/start
2. While logged in or as a logged out (new) user, go through the new site flow
3. On the domains step, enter a top level domain that you have registered elsewhere - in my case, this was a domain registered at GoDaddy that I had planned on mapping to a new WordPress.com site.
4. Notice that the domain validation displays "domain available"
5. Continue with registration for the domain - note no errors
6. Try to checkout with domain in cart
7. Submit payment without validation errors
8. Get error after submitting payment that domain "exists and cannot be registered"

![](https://cloudup.com/cogavQAJ4qH+)

cc @aidvu @klimeryk

**Comment by aidvu** [comment]
Very nasty bug. Investigated a minute ago. The problem here is that if the domain is in the suggestions list, it will be shown as available, w/o checking the availability results.

**Comment by aidvu** [comment]
Not that nasty. Misunderstood something. 👊

**Comment by aidvu** [comment]
Fix: https://github.com/Automattic/wp-calypso/pull/7896

**Comment by klimeryk** [comment]
Fixed by #7896 🎉
